apps/prairielearn/src/pages/home/home.tsx

```diff
@@ -121,15 +128,45 @@ router.post(
     } = getPageContext(res.locals, { withAuthzData: false });
 
     if (body.__action === 'accept_invitation') {
-      await execute(sql.accept_invitation, {
+      await ensureEnrollment({
         course_instance_id: body.course_instance_id,
-        uid,
         user_id,
+        agent_user_id: user_id,
+        agent_authn_user_id: user_id,
+        action_detail: 'invitation_accepted',
       });
     } else if (body.__action === 'reject_invitation') {
-      await execute(sql.reject_invitation, {
-        course_instance_id: body.course_instance_id,
-        uid,
+      await runInTransactionAsync(async () => {
+        const oldEnrollment = await selectOptionalEnrollmentByPendingUid({
+          course_instance_id: body.course_instance_id,
+          pending_uid: uid,
+        });
+
+        if (!oldEnrollment) {
+          throw new Error('Could not find enrollment to reject');
+        }
+        await selectAndLockEnrollmentById(oldEnrollment.id);
```

Comment:
In other places you lock the user before locking the enrollment; here you do not. If there's a reason for the inconsistency, please document it; if not, please add a lock.

Reply:
I updated the lock again in c0add6bacf9a32aed633d9d75e2ffd537bfce788.